bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/accessories/HomekitThermostatImpl.java

```diff
@@ -50,219 +46,207 @@
+                    stringValue, getSettings().thermostatCurrentModeCooling, getSettings().thermostatCurrentModeHeating,
+                    getSettings().thermostatCurrentModeOff);
+            mode = CurrentHeatingCoolingStateEnum.OFF;
         }
-
-        characteristicItems.entrySet().stream().forEach(entry -> {
-            logger.warn("Item {} has unrecognized thermostat characteristic: {}", entry.getValue().getUID(),
-                    entry.getKey().getTag());
-        });
+        return CompletableFuture.completedFuture(mode);
     }
 
-    private Optional<Item> getItemWithDeprecation(HashMap<HomekitCharacteristicType, Item> characteristicItems,
-            HomekitCharacteristicType currentTag, HomekitCharacteristicType deprecatedTag) {
-        Optional<Item> targetTempItem = Optional.ofNullable(characteristicItems.remove(currentTag));
-        if (!targetTempItem.isPresent()) {
-            targetTempItem = Optional.ofNullable(characteristicItems.remove(deprecatedTag));
-            targetTempItem.ifPresent(item -> {
-                logger.warn("The tag {} has been renamed to {}; please update your things, accordingly",
-                        deprecatedTag.getTag(), currentTag.getTag());
-            });
-        }
-        return targetTempItem;
+    @Override
+    public CompletableFuture<Double> getCurrentTemperature() {
+        DecimalType state = getStateAs(HomekitCharacteristicType.CURRENT_TEMPERATURE, DecimalType.class);
+        return CompletableFuture.completedFuture(state != null ? convertToCelsius(state.doubleValue()) : 0.0);
     }
 
     @Override
-    public CompletableFuture<ThermostatMode> getCurrentMode() {
-        String stringValue = settings.thermostatCurrentModeOff;
-        if (currentHeatingCoolingModeItem != null) {
-            stringValue = currentHeatingCoolingModeItem.getState().toString();
+    public CompletableFuture<TargetHeatingCoolingStateEnum> getTargetState() {
+        String stringValue = getSettings().thermostatTargetModeOff;
+
+        final Optional<HomekitTaggedItem> characteristic = getCharacteristic(
+                HomekitCharacteristicType.TARGET_HEATING_COOLING_STATE);
+        if (characteristic.isPresent()) {
+            stringValue = characteristic.get().getItem().getState().toString();
+        } else {
+            logger.warn("Missing mandatory characteristic {}", HomekitCharacteristicType.TARGET_HEATING_COOLING_STATE);
         }
-        ThermostatMode mode;
-
-        if (stringValue.equalsIgnoreCase(settings.thermostatCurrentModeCooling)) {
-            mode = ThermostatMode.COOL;
-        } else if (stringValue.equalsIgnoreCase(settings.thermostatCurrentModeHeating)) {
-            mode = ThermostatMode.HEAT;
-        } else if (stringValue.equalsIgnoreCase(settings.thermostatCurrentModeOff)) {
-            mode = ThermostatMode.OFF;
+        TargetHeatingCoolingStateEnum mode;
+
+        if (stringValue.equalsIgnoreCase(getSettings().thermostatTargetModeCool)) {
```

Comment:
I would prefer if you cached `getSettings()` in a local variable to prevent calling it excessively.